app/models/crop.rb

```diff
@@ -52,11 +52,19 @@ class Crop
   has_many :varieties, class_name: 'Crop', inverse_of: :parent
   belongs_to :parent, class_name: 'Crop', inverse_of: :varieties
 
+  has_and_belongs_to_many :companions,
+                          class_name: 'Crop',
+                          inverse_of: :companions
+
```

Comment:
@simonv3 I haven't been tracking the `Companion` discussion as of late, so I might be missing details, but it sounds similar to [what this SA answer suggests](http://stackoverflow.com/a/7497739/1064917):

```ruby
class Link
  has_and_belongs_to_many :companions,
    :class_name => 'Crop',
    :inverse_of => :suggested_companions
  has_and_belongs_to_many :suggested_companions,
    :class_name => 'Crop',
    :inverse_of => :companions
end
```

The description in the link sounds spot on:

  > you need to give mongoid a little help with the class_name and inverse_of because it can't infer this from the association name.